### PR TITLE
Echo participant [12947]

### PR DESCRIPTION
**Pull request by jparisu**
Merge after #37

### CODE REVIEW

File: src/cpp/types/ParticipantType.cpp [hunk 1]
```diff
@@ -36,6 +36,11 @@ ParticipantType ParticipantTypeFactory::participant_type_from_name(
         // Void type
         return ParticipantType::VOID;
     }
+    else if (type == ECHO_TYPE_NAME)
+    {
+        // Void type
```
Comment (juanlofer-eprosima) [inline src/cpp/types/ParticipantType.cpp:41]:
```suggestion
        // Echo type
```

File: src/cpp/writer/implementations/auxiliar/EchoWriter.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+void EchoWriter::disable()
+{
+    enabled_.store(false);
+}
+
+ReturnCode EchoWriter::write(std::unique_ptr<DataReceived>& data)
+{
+    if (enabled_.load())
+    {
+        std::cout << "Echo Participant: " << participant_id_ << " has recived from Endpoint: " << data->source_guid
```
Comment (juanlofer-eprosima) [inline src/cpp/writer/implementations/auxiliar/EchoWriter.cpp:47]:
```suggestion
        std::cout << "Echo Participant: " << participant_id_ << " has received from Endpoint: " << data->source_guid
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Please fix uncrustify test

File: include/ddsrouter/participant/implementations/auxiliar/EchoParticipant.hpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * @file EchoParticipant.hpp
+ */
+
+#ifndef _DDSROUTER_PARTICIPANT_IMPLEMENTATIONS_AUX_ECHOPARTICIPANT_HPP_
+#define _DDSROUTER_PARTICIPANT_IMPLEMENTATIONS_AUX_ECHOPARTICIPANT_HPP_
+
+#include <ddsrouter/participant/implementations/auxiliar/VoidParticipant.hpp>
+#include <ddsrouter/configuration/ParticipantConfiguration.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/participant/implementations/auxiliar/EchoParticipant.hpp:23]: :+1:

File: include/ddsrouter/participant/implementations/auxiliar/VoidParticipant.hpp [hunk 4]
```diff
@@ -20,6 +20,7 @@
 #define _DDSROUTER_PARTICIPANT_IMPLEMENTATIONS_AUX_VOIDPARTICIPANT_HPP_
 
 #include <ddsrouter/participant/IParticipant.hpp>
+#include <ddsrouter/configuration/ParticipantConfiguration.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/participant/implementations/auxiliar/VoidParticipant.hpp:23]: :+1:

File: include/ddsrouter/types/ParticipantType.hpp [hunk 5]
```diff
@@ -31,6 +31,7 @@ enum ParticipantType
 {
     INVALID,    //! Invalid Participant Type
     VOID,       //! Void Participant Type
+    ECHO,       //! Void Participant Type
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/ParticipantType.hpp:34]: :+1:

File: include/ddsrouter/writer/implementations/auxiliar/EchoWriter.hpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ * @file EchoWriter.hpp
+ */
+
+#ifndef _DDSROUTER_WRITER_IMPLEMENTATIONS_AUX_ECHOWRITER_HPP_
+#define _DDSROUTER_WRITER_IMPLEMENTATIONS_AUX_ECHOWRITER_HPP_
+
+#include <atomic>
+
+#include <ddsrouter/writer/IWriter.hpp>
+#include <ddsrouter/types/ParticipantId.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/writer/implementations/auxiliar/EchoWriter.hpp:25]: :+1:

File: src/cpp/participant/implementations/auxiliar/EchoParticipant.cpp [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+// limitations under the License.
+
+/**
+ * @file EchoParticipant.cpp
+ */
+
+#include <ddsrouter/participant/implementations/auxiliar/EchoParticipant.hpp>
+#include <ddsrouter/reader/implementations/auxiliar/VoidReader.hpp>
+#include <ddsrouter/writer/implementations/auxiliar/EchoWriter.hpp>
+#include <ddsrouter/types/ParticipantType.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/participant/implementations/auxiliar/EchoParticipant.cpp:22]: :+1:

File: src/cpp/types/ParticipantType.cpp (hunk 1 above)
Comment (jparisu) [reply src/cpp/types/ParticipantType.cpp:41]: :+1:

File: src/cpp/writer/implementations/auxiliar/EchoWriter.cpp (hunk 2 above)
Comment (jparisu) [reply src/cpp/writer/implementations/auxiliar/EchoWriter.cpp:47]: :+1:

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Please fix windows build

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**